ocs_ci/ocs/resources/bucket_notifications_manager.py

```diff
@@ -0,0 +1,360 @@
+            bucket_notifs_dict["pvc"] = provided_notifs_pvc
+
+        patch_params = [
+            {
+                "op": "add",
+                "path": NOTIFS_YAML_PATH_NB_CR,
+                "value": bucket_notifs_dict,
+            }
+        ]
+
+        # Try patching via add, and if it fails - replace instead
+        try:
+            self.nb_config_resource.patch(
+                params=json.dumps(patch_params),
+                format_type="json",
+            )
+        except CommandFailed as e:
+            if "already exists" in str(e).lower():
+                patch_params[0]["op"] = "replace"
+                self.nb_config_resource.patch(
+                    params=json.dumps(patch_params),
+                    format_type="json",
+                )
+            else:
+                logger.error(f"Failed to enable bucket notifications: {e}")
+                raise e
+
+        # Label the PVC and PV to tolerate as leftovers after the test
+        pvc_name = provided_notifs_pvc or constants.DEFAULT_MCG_BUCKET_NOTIFS_PVC
+        pvc_ocp_obj = OCP(
+            namespace=self.namespace,
+            kind="pvc",
+            resource_name=pvc_name,
+        )
+        pv_name = pvc_ocp_obj.get()["spec"]["volumeName"]
+        pv_ocp_obj = OCP(
+            namespace=self.namespace,
+            kind="pv",
+            resource_name=pv_name,
+        )
+        for ocp_obj in (pvc_ocp_obj, pv_ocp_obj):
+            ocp_obj.add_label(
+                resource_name=ocp_obj.resource_name,
+                label=constants.CUSTOM_MCG_LABEL,
+            )
+        wait_for_pods_to_be_running(
+            pod_names=[constants.NOOBAA_CORE_POD],
+            timeout=60,
+            sleep=10,
+        )
+        logger.info("Guaranteed bucket logs have been enabled")
+
+    def disable_bucket_logging_on_cr(self):
+        """
+        Unset the bucket notifications feature on the NooBaa CR
+        """
+        logger.info("Disabling bucket notifications on the NooBaa CR")
+
+        try:
+            patch_params = [
+                {
+                    "op": "replace",
+                    "path": NOTIFS_YAML_PATH_NB_CR,
+                    "value": None,
+                },
+            ]
+            self.nb_config_resource.patch(
+                params=json.dumps(patch_params),
+                format_type="json",
+            )
+        except CommandFailed as e:
+            if "not found" in str(e).lower():
+                logger.info("The bucketNotifications field was not found")
+            else:
+                logger.error(f"Failed to disable bucket notifications: {e}")
+                raise e
+        wait_for_pods_to_be_running(
+            pod_names=[constants.NOOBAA_CORE_POD],
+            timeout=60,
+            sleep=10,
+        )
+        logger.info("Bucket notifications have been disabled")
+
+    def create_kafka_topic(self, topic_name=""):
+        """
+        Create a Kafka topic
```

Comment:
please match the Args to the actual argument name

Reply:
Fixed in https://github.com/red-hat-storage/ocs-ci/pull/11041/commits/d947ebdb6635fc66de628791628bd1f1089b5e96